skimage/feature/tests/test_hog.py

```diff
@@ -287,3 +287,16 @@ def test_hog_output_equivariance_channel_axis(channel_axis):
         hog_fact = feature.hog(np.roll(img, n, axis=channel_axis),
                                channel_axis=channel_axis, block_norm='L1')
         assert_almost_equal(hog_ref, hog_fact)
+
+
+def test_hog_small_image():
+    """
+    tests that an exception is thrown instead of an empty array returned when
+    the image is smaller than it should be for the given parameters
+    """
+    img = np.zeros((24, 193))
+    feature.hog(img, pixels_per_cell=(8, 8), cells_per_block=(3, 3))
```

Comment:
Could you use the minimal required image size here? That way we can be sure that future changes won't affect the edge case. If you want you could even pass in different `pixels_per_cell` and `cells_per_block` with `@pytest.mark.parametrize` and calculate the passing and failing image size on the fly...